CMakeLists.txt

```diff
@@ -95,6 +95,7 @@ if(BASEARCH_ARM_FOUND)
     add_option(WITH_ACLE "Build with ACLE" ON)
     add_option(WITH_NEON "Build with NEON intrinsics" ON)
 elseif(BASEARCH_PPC_FOUND)
+    add_option(WITH_VMX "Build with VMX optimisations for PowerPC" ON)
     add_option(WITH_POWER8 "Build with optimisations for POWER8" ON)
```

Comment:
Unrelated to this PR, but I am wondering whether the POWER8 options should perhaps have had a different name, one more in line with the rest (naming instruction sets, not cpu generation). Generally that allows more fine-grained control too. WITH_VSX might have been more appropriate.
Not sure whether we discussed this already.

Reply:
@Dead2 I was already thinking about changing so that both configure and cmake use "VMX" instead of mixing "AltiVec" and "VMX" even though in the headers, the feature bit is still called "AltiVec". However "AltiVec" is kind of the instruction set as a whole (VMX+VSX). There is only two relevant instructions that are missing in VMX/AltiVec (can be emulated with extend-multiply-even and extend-multiply-odd followed by addition).

CodeCov seems to be having some issues with VMX code even though I disabled VSX support in one of the CI runs... I haven't investigated yet how CodeCov gathers all the data for the two failing tests.

Reply:
Another name for VMX/AltiVec is "Velocity Engine"... VMX itself is abbreviation of "Vector Multimedia Extension" and VSX is "Vector Scalar Extension"... VSX is not completely separate instruction set, as at the same time when some VSX instructions were added. also some instructions were added to VMX.

As mentioned, there is a lot of conflicting naming used for the vector instructions, so referring to specific ISA version at least internally makes more sense. That is also how the feature detection is currently implemented in `arch/power/power.c`  of zlib-ng.

Reply:
I changed that both cmake and configure use name "AltiVec" to match the feature flag in `getauxval`...